### PR TITLE
Support for upsert

### PR DESCRIPTION
**Pull request by nirbhayc**

### CODE REVIEW

File: db/record.c [hunk 1]
```diff
@@ -461,7 +521,7 @@ int add_record(struct ireq *iq, void *trans, const uint8_t *p_buf_tag_name,
         if (!(flags & RECFLAGS_NEW_SCHEMA)) {
             /* enqueue the add of the key for constaint checking purposes */
             rc = insert_add_op(iq, iq->blkstate, iq->usedb, NULL, NULL, opcode,
-                               *rrn, -1, *genid, ins_keys, blkpos);
+                               *rrn, -1, *genid, ins_keys, blkpos, rec_flags);
```
Comment (nirbhayc) [inline db/record.c]: we dont need rec_flags for delay_key_add

File: db/record.c [hunk 2]
```diff
@@ -399,6 +400,65 @@ int add_record(struct ireq *iq, void *trans, const uint8_t *p_buf_tag_name,
         ERR;
     }
 
+    if ((rec_flags & OSQL_IGNORE_FAILURE) != 0) {
+        for (int ixnum = 0; ixnum < iq->usedb->nix; ixnum ++) {
```
Comment (nirbhayc) [inline db/record.c]: BUG: check only unique keys in upsert targets

**Comment by nirbhayc** [comment]
Thanks @ldennis !

**Comment by ldennis** [comment]
TODO: Documentation and Query Effects

**Pull request merged**